geode-core/src/main/java/org/apache/geode/internal/net/NioSslEngine.java

```diff
@@ -52,6 +53,11 @@
 public class NioSslEngine implements NioFilter {
   private static final Logger logger = LogService.getLogger();
 
+  // this variable requires the MakeImmutable annotation but the buffer is empty and
+  // not really modifiable
+  @MakeImmutable
+  private static final ByteBuffer EMPTY_BUFFER = ByteBuffer.allocate(0);
```

Comment:
did you consider adding ".asReadOnlyBuffer()" so that this would be immutable?

Reply:
That won't satisfy PMD (since it returns a `ByteBuffer` which is not considered an immutable object by PMD) but it still might be valuable.